SeQuant/domain/eval/eval.hpp

```diff
@@ -344,6 +344,7 @@ auto evaluate(NodesT const& nodes, Le const& le, Args&&... args) {
 /// \see EvalResult to know more about the return type.
 ///
 template <typename NodeT, typename Annot, typename Le, typename... Args,
+          std::enable_if_t<IsEvaluable<NodeT>, bool> = true,
           std::enable_if_t<IsLeafEvaluator<NodeT, Le>, bool> = true>
```

Comment:
`std::enable_if_t<CondA>, std::enable_if_t<CondB>` is more compactly expressed as `std::enable_if_t<CondA && CondB>`

Reply:
Separating these checks allows identifying the offending check by looking at the compiler error messages when no candidate function is found. That is, in this case, whether `IsEvaluable<NodeT>` failed or `IsLeafEvaluator<NodeT, Le>` failed. Combined, the error message does not tell which check failed. (PS. Error messages articulation might be a compiler specific behavior.)